ide/utils/regexes.py

```diff
@@ -0,0 +1,35 @@
+from django.core.validators import RegexValidator
+
+
+class RegexHolder(object):
+    regex_dictionary = {
+        # Match major[.minor], where major and minor are numbers between 0 and 255 with no leading 0s
+        'sdk_version': r'^(0|[1-9]\d?|1\d{2}|2[0-4]\d|25[0-5])(\.(0|[1-9]\d?|1\d{2}|2[0-4]\d|25[0-5]))?$',
+
+        # Match major.minor.0, where major and minor are numbers between 0 and 255 with no leading 0s
+        'semver': r'^(0|[1-9]\d?|1\d{2}|2[0-4]\d|25[0-5])\.(0|[1-9]\d?|1\d{2}|2[0-4]\d|25[0-5])\.0$',
+
+        # Match a string of letters and numbers separated with underscores but not starting with a digit
+        'c_identifier': r'^([a-zA-Z_][_a-zA-Z\d]*)$',
```

Comment:
More broadly, every use of this has the same property: it has a prefix when used in C, and so actually can start with a digit.